examples/ui/text_cursor.rs

```diff
@@ -0,0 +1,109 @@
+//! This example illustrates how to create and control a UI text cursor
+
+use bevy::{
+    color::palettes::css::GOLDENROD,
+    diagnostic::FrameTimeDiagnosticsPlugin,
+    prelude::*,
+    ui::widget::{TextCursor, TextCursorStyle, TextCursorWidth},
+};
+
+const CURSOR_WIDTH: f32 = 4.;
+
+fn main() {
+    App::new()
+        .add_plugins((DefaultPlugins, FrameTimeDiagnosticsPlugin::default()))
+        .add_systems(Startup, setup)
+        .add_systems(Update, move_cursor)
+        .run();
+}
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    // UI camera
+    commands.spawn(Camera2d);
+    // Text with one section
+    commands
+        .spawn(Node {
+            width: Val::Percent(100.),
+            height: Val::Percent(100.),
+            justify_content: JustifyContent::Center,
+            align_items: AlignItems::Center,
+            row_gap: Val::Px(50.),
+            flex_direction: FlexDirection::Column,
+            ..Default::default()
+        })
+        .with_child((
+            // Accepts a `String` or any type that converts into a `String`, such as `&str`
+            Text::new(
+                "Lorem ipsum dolor sit amet,\nconsectetur adipiscing elit,\nsed do eiusmod tempor incididunt\nut labore et dolore magna aliqua."
+            ),
+            TextFont {
+                // This font is loaded and will be used instead of the default font.
+                font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                font_size: 30.0,
+                ..default()
+            },
+            TextColor(GOLDENROD.into()),
+            TextCursor {
+                line: 0,
+                index: 0,
+            },
+            // Set the justification of the Text
+            TextLayout::new_with_justify(JustifyText::Center),
+            // Set the style of the Node itself.
+            Outline {
+                color: Color::WHITE,
+                width: Val::Px(2.),
+                offset: Val::Px(25.),
+            },
+        ))
+        .with_child((
+            // Accepts a `String` or any type that converts into a `String`, such as `&str`
+            Text::new(
+                "Lorem ipsum dolor sit amet,\nconsectetur adipiscing elit,\nsed do eiusmod tempor incididunt\nut labore et dolore magna aliqua."),
+            TextFont {
+                // This font is loaded and will be used instead of the default font.
+                font: asset_server.load("fonts/FiraMono-Medium.ttf"),
```

Comment:
Do we need this whole font or will the default font do?

Reply:
Yeah just the default is fine, it's just to contrast mono and variable width fonts.